src/Services/Qdm/Services/AbstractCarePlanService.php

```diff
@@ -31,11 +33,41 @@ abstract class AbstractCarePlanService extends AbstractQdmService
 
     abstract public function getCarePlanType();
 
+    abstract public function getModelClass();
+
     public function getSqlStatement()
     {
         $carePlanType = $this->getCarePlanType();
         return "SELECT pid, `date`, code, codetext, description, care_plan_type, reason_code
             FROM form_care_plan
             WHERE care_plan_type = '" . add_escape_custom($carePlanType) . "'";
     }
+
+    /**
+     * @param array $record
+     * @return mixed
+     * @throws \Exception
+     *
+     * Since almost all the care plans contain the same data, we put the base code for making the model here.
+     */
+    public function makeQdmModel(array $record)
+    {
+        $modelClass = $this->getModelClass();
+        $model = new $modelClass([
+            'authorDatetime' => new DateTime([
+                'date' => $record['date']
+            ]),
+        ]);
+
+        $model->addCode($this->makeQdmCode($record['code']));
+
+        // If there is a reason noted why this plan was NOT done, add a negation
+        if ($this->isNegationCode($record['reason_code'])) {
+            $model->negationRationale = $this->makeQdmCode($record['reason_code']);
```

Comment:
doesn't this need to be $this->isNegationCode($record['code'] because code is what is encoded with OID

Reply:
I understood that the "code" column had the "what was done" and the "reason_code" column had the "why, or why not" code. So, the "OID:x.xxx.xxx.xx" should be in the reason_code because it's the reason why code "some code" wasn't done. 
![Screen Shot 2022-04-11 at 10 26 19 AM](https://user-images.githubusercontent.com/271809/162761587-84ebe822-86bd-433b-8d7b-dcdabd2956ba.png)

Reply:
Nope it's the code that gets nulled and set to the valueset.
![image](https://user-images.githubusercontent.com/1859985/162762266-a501e49f-ff5d-4dcc-ae9d-386835e3fcd2.png)

Reply:
![image](https://user-images.githubusercontent.com/1859985/162764691-946ae39a-5442-448e-85d3-45e51d086506.png)